File: cases/taint/javascript/loop-carried-negative/LoopCarriedNegative.js
function dfb_source() { // DFB-SOURCE: loop-carried-input
  return 1;
}

function dfb_sink(value) {} // DFB-SINK: loop-carried-sink

function run() {
  let value = dfb_source();
  for (let iteration = 0; iteration < 3; iteration++) {
    value = 0; // DFB-WITNESS: loop-carried-value
  }
  dfb_sink(value);
}

File: cases/taint/go/loop-carried-negative/loop_carried_negative.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: loop-carried-input
	return 1
}

func dfb_sink(value int) {} // DFB-SINK: loop-carried-sink

func run() {
	value := dfb_source()
	for iteration := 0; iteration < 3; iteration++ {
		value = 0 // DFB-WITNESS: loop-carried-value
	}
	dfb_sink(value)
}
